sys/net/gnrc/link_layer/lwmac/lwmac.c

```diff
@@ -743,83 +875,79 @@ static void *_lwmac_thread(void *args)
-    gnrc_lwmac_reset_timeouts(gnrc_netdev);
+    gnrc_lwmac_reset_timeouts(netif);
 
     /* Start duty cycling */
-    lwmac_set_state(gnrc_netdev, GNRC_LWMAC_START);
+    lwmac_set_state(netif, GNRC_LWMAC_START);
 
 #if (GNRC_LWMAC_ENABLE_DUTYCYLE_RECORD == 1)
     /* Start duty cycle recording */
-    gnrc_netdev->lwmac.system_start_time_ticks = rtt_get_counter();
-    gnrc_netdev->lwmac.last_radio_on_time_ticks = gnrc_netdev->lwmac.system_start_time_ticks;
-    gnrc_netdev->lwmac.awake_duration_sum_ticks = 0;
-    gnrc_netdev->lwmac.lwmac_info |= GNRC_LWMAC_RADIO_IS_ON;
+    netif->mac.lwmac.system_start_time_ticks = rtt_get_counter();
+    netif->mac.lwmac.last_radio_on_time_ticks = netif->mac.lwmac.system_start_time_ticks;
+    netif->mac.lwmac.awake_duration_sum_ticks = 0;
+    netif->mac.lwmac.lwmac_info |= GNRC_LWMAC_RADIO_IS_ON;
 #endif
 
     /* start the event loop */
     while (1) {
         msg_receive(&msg);
-
-        /* Handle NETDEV, NETAPI, RTT and TIMEOUT messages */
+        printf("gnrc_netif2: get incoming messages\n");
         switch (msg.type) {
-            /* RTT raised an interrupt */
-            case GNRC_LWMAC_EVENT_RTT_TYPE: {
-                if (gnrc_netdev_lwmac_get_dutycycle_active(gnrc_netdev)) {
-                    rtt_handler(msg.content.value, gnrc_netdev);
-                    lwmac_schedule_update(gnrc_netdev);
-                }
-                else {
-                    LOG_DEBUG("[LWMAC] Ignoring late RTT event while dutycycling is off\n");
-                }
-                break;
-            }
-            /* An LWMAC timeout occured */
-            case GNRC_LWMAC_EVENT_TIMEOUT_TYPE: {
-                gnrc_lwmac_timeout_make_expire((gnrc_lwmac_timeout_t *) msg.content.ptr);
-                lwmac_schedule_update(gnrc_netdev);
-                break;
-            }
             /* Transceiver raised an interrupt */
             case NETDEV_MSG_TYPE_EVENT: {
-                LOG_DEBUG("[LWMAC] GNRC_NETDEV_MSG_TYPE_EVENT received\n");
-                /* Forward event back to driver */
+                DEBUG("[LWMAC] GNRC_NETDEV_MSG_TYPE_EVENT received\n");
                 dev->driver->isr(dev);
                 break;
             }
+
             /* TX: Queue for sending */
             case GNRC_NETAPI_MSG_TYPE_SND: {
                 /* TODO: how to announce failure to upper layers? */
                 LOG_DEBUG("[LWMAC] GNRC_NETAPI_MSG_TYPE_SND received\n");
                 gnrc_pktsnip_t *pkt = (gnrc_pktsnip_t *) msg.content.ptr;
 
-                if (!gnrc_mac_queue_tx_packet(&gnrc_netdev->tx, 0, pkt)) {
+                if (!gnrc_mac_queue_tx_packet(&netif->mac.tx, 0, pkt)) {
                     gnrc_pktbuf_release(pkt);
                     LOG_WARNING("WARNING: [LWMAC] TX queue full, drop packet\n");
                 }
 
-                lwmac_schedule_update(gnrc_netdev);
+                lwmac_schedule_update(netif);
                 break;
```

Comment:
This all can go into `gnrc_netif2_ops_t::send()` so you don't need the thread anymore.